### PR TITLE
Revert #46734: OpenAPI aggregation for kube-aggregator

### PR DESCRIPTION
**Pull request by sttts**
As discussed in #46734 this PR reverts the code in master for 1.8.

### CODE REVIEW

File: pkg/generated/openapi/BUILD [hunk 1]
```diff
@@ -8,6 +8,10 @@ openapi_library(
     name = "go_default_library",
     srcs = ["doc.go"],
     openapi_targets = [
+        "_output/dockerized/go/src/github.com/heketi/heketi/vendor/k8s.io/kubernetes/pkg/runtime",
```
Comment (sttts) [inline pkg/generated/openapi/BUILD]: Bleh, why do I get this?

**Comment by deads2k** [comment]
/lgtm


@sttts open the reimplementation issue for @mbohlool ?

**Comment by sttts** [comment]
@smarterclayton approved?

**Comment by mbohlool** [comment]
Do we want to revert it or just send more PRs to fix it? @lavalamp

**Comment by lavalamp** [comment]
If we aren't reverting in 1.7, what is the rush? Why not just fix it?

**Comment by lavalamp** [comment]
I think I would greatly prefer the revert and fix to come in the same PR, otherwise we have a huge regression in 1.8 right off the bat.

**Comment by mbohlool** [comment]
I second that, specially because we may not need to revert everything. For example I don't see any problem with "Remove unused servePath from GetOperationIDAndTags..." commit or ""Enable OpenAPI definition generation for apiregistration" one.

**Comment by deads2k** [comment]
> I think I would greatly prefer the revert and fix to come in the same PR, otherwise we have a huge regression in 1.8 right off the bat.

I don't think its huge.  The two know UASs work fine without it working in head.  And in fact the current code doesn't even work OOTB with the existing types (CFD is missing).

Given the breadth of structural problems in the initial pull, I don't have confidence in a piecemeal refactor.  I'd like to see the individual steps worked back up with a more thorough review at each step.

**Comment by mbohlool** [comment]
I didn't get your point about current code does not work.

We can still have what you are asking in a single PR. What is the benefit of reverting without a fix in place?

**Comment by deads2k** [comment]
> We can still have what you are asking in a single PR. What is the benefit of reverting without a fix in place?

It forces proper structure on the replacement.  There are a series of logical steps to get from A to B and by starting over with a clean slate you can go through them.  It will make easier to write and easier to review.  Otherwise we're chasing spaghetti around trying to remember which pieces still have to move.

In addition, like the resistance being faced here, "it kind of works, let me stick a few fingers in the dike" is really tempting if something sort of half exists.  We don't want this structure affecting future structure and reverting it ensures we don't end up there.

**Comment by mbohlool** [comment]
The logical steps can still happened in one PR. A commit to revert follow by commits to reintegrate.

I would have agreed with you on second point if the feature were highly developed by many parties but this Spec integration would not be likely in active development before wr fix its structure.

I understand you guys want the restructuring to happen but the right way to do do is to increase priority on the related issue.

**Comment by deads2k** [comment]
> I understand you guys want the restructuring to happen but the right way to do do is to increase priority on the related issue.

Past experience indicates that this is very unlikely to happen if we do that.  Consider https://github.com/kubernetes/kubernetes/issues/41748 as an example.  Other than claiming its actively failing something, the priority doesn't get higher.

Reversion will ensure completion in a way that an issue cannot.

**Comment by lavalamp** [comment]
I'm taking the lgtm off while @deads2k and I hash this out in a higher-bandwidth medium.

**Comment by gmarek** [comment]
As for some reason I have this PR in my stream I'll add my very generic 2c - that I agree with @deads2k: I think we don't revert enough. I believe that we'd end up with much healthier codebase if we would by default just revert everything that a) breaks something and needs to be fixed and b) it's not hard to revert. It'd increase the incentive for pushing better code, as if you push something problematic, you'll need to do, possibly hard, rebase after fix is ready.

**Comment by lavalamp** [comment]
The outcome of our conversation is that we (well, @mbohlool) will prioritize changing this and hope to get something out next week.

As far as reverts go, I think stuff that breaks tests can be reverted unconditionally. I don't think I want to live in a world where my changes can be reverted for (m)any other reasons, though. It's anxiety-inducing enough to get changes in in the first place. People need closure. I think we should save direct reverts for cases where there's clear malfeasance; in the ordinary case, we should trust authors to fix things if they say they will, etc. And if timelines don't match up, the historical precedent in this project is that the folks with the more pressing timeline end up doing more work. It's not wonderful but we shouldn't try to change it without a broader discussion.

In this particular case we were in a rush; @deads2k and I agreed that going forward we don't anticipate situations like this again and we will just block PRs instead of letting in things that we know we want to change.

For this particular PR, there's some debate about how much of it needs to change and I think we will work that out in the coming PR. David and I need to do a better job of documenting the theory behind each of apimachinery's repos, so that we don't have to figure this stuff out on the fly.

**Comment by sttts** [comment]
Replaced by https://github.com/kubernetes/kubernetes/pull/48861

**Pull request closed without merging**